src/H5private.h

```diff
@@ -1385,6 +1386,9 @@ H5_DLL H5_ATTR_CONST int Nflock(int fd, int operation);
 #ifndef HDstrcat
 #define HDstrcat(X, Y) strcat(X, Y)
 #endif
+#ifndef HDstrcasestr
+#define HDstrcasestr(X, Y) H5strcasestr(X, Y)
+#endif
```

Comment:
This would use  H5_strcasestr() on all platforms.

What you want to do is check for strcasestr in the configure files (both Autotools and CMake) and then the code here will look something like:

```
#ifndef HDstrcasestr
#if defined(H5_HAVE_STRCASESTR)                                                       
#define HDstrcasestr(X, Y) strcasestr(X, Y)                                                                                     
#else                                                                            
#define HDstrcasestr(X, Y) H5_strcasestr(X, Y)                                              
#endif
```
This defines the macro to be the C library call if we found it or the new private function if not (both only in the case that we didn't do something else in H5win32defs.h or some other OS-specific file in the future).

Reply:
Also, ideally, you'd want to #define HDstrcasestr() to StlStrA or whatnot in H5win32defs.h so this call will work on Windows.

Reply:
For the Autotools, look in configure.ac and add it to the block of AC_CHECK_FUNCS (note that they are in alphabetical order).

Reply:
For CMake, add it to the CHECK_FUNCTION_EXISTS section in config/cmake/ConfigureChecks.cmake (around line 450) and also add a section to config/cmake/H5pubconf.h.in that matches the Autotools output (we try to keep the pubconf headers identical between the Autotools and CMake).

Reply:
If strcasestr were POSIX, we'd probably decorate the replacement function with a 'W' (Wstrcasestr), not do the function tests, and only replace on Windows, but since it's "nonstandard" instead of POSIX we have to worry about it not being present on uncommon Unix-y platforms and the H5_strcasestr() scheme w/ configure checks is the right way to go.